src/v2/guide/components.md

```diff
@@ -434,6 +434,44 @@ In addition, `type` can also be a custom constructor function and the assertion
 
 When a prop validation fails, Vue will produce a console warning (if using the development build).
 
+## Non-Prop Attributes
+
```

Comment:
👋 Hi, vue user here and followed to this pr from the issue discussion vuejs repo.

Have you considered adding a bottom line up front statement of the behavior and then the detail below that?  For instance, maybe a sentence here with something like.

```
A non-prop attribute is an attribute that is passed to a component that does not have a corresponding prop defined.

Non-prop attributes are passed as attributes to the root element of the component they are sent to.
```